src/Duende.AccessTokenManagement.OpenIdConnect/IPrincipalAccessor.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Brock Allen & Dominick Baier. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Duende.AccessTokenManagement.OpenIdConnect;
+
+/// <summary>
+/// Service that retrieves the current principal.
+/// </summary>
+public interface IPrincipalAccessor
+{
+    /// <summary>
+    /// Gets the current user.
+    /// </summary>
+    Task<ClaimsPrincipal> GetCurrentUserAsync();
```

Comment:
I know naming is hard... "principal" accessor, and get current "user". Any thoughts on the discrepancy, or are you ok with those names?

Reply:
I don't love the names, but I think I prefer User to Principal because it is more ... I don't know, concrete?

So rename to:
IUserAccessor
HttpContextUserAccessor
BlazorServerUserAccessor

Reply:
I'm ok with anything... I just wanted you to give it one more round of thinking. So I'm fine with these new names you are proposing.